scripts/dependency-graph.js

```diff
@@ -1,189 +1,153 @@
-  if (name.startsWith('@')) {
-    const parts = name.slice(1).split('/')
-    return `${parts[0]}-${parts[1]}`
-  }
-  return name
+function stripName(name) {
+  return name.startsWith("@")
+    ? `${name.slice(1).split("/")[0]}-${name.slice(1).split("/")[1]}`
+    : name;
 }
 
-const main = async function () {
+async function main() {
   // add all of the cli's public workspaces as package names
-  for (const { name, pkg: ws } of await pkg.mapWorkspaces()) {
-    if (!ws.private) {
-      repos.push(name)
-    }
-  }
-
-  const arborist = new Arborist({ prefix: CWD, path: CWD })
-  const tree = await arborist.loadVirtual({ path: CWD, name: 'npm' })
-  tree.name = 'npm'
+  const arborist = new Arborist({ prefix: __dirname, path: __dirname });
+  const tree = await arborist.loadVirtual({ path: __dirname, name: "npm" });
+  tree.name = "npm";
 
-  const [annotationsOurs, hierarchyOurs] = walk(tree, true)
-  const [annotationsAll] = walk(tree, false)
+  const [annotationsOurs, hierarchyOurs] = walk(tree, true);
+  const [annotationsAll] = walk(tree, false);
 
   const out = [
-    '# npm dependencies',
-    '',
-    '## `github.com/npm/` only',
-    '```mermaid',
-    'graph LR;',
+    "# npm dependencies",
+    "",
+    "## `github.com/npm/` only",
+    "```mermaid",
+    "graph LR;",
     ...annotationsOurs.sort(),
-    '```',
-    '',
-    '## all dependencies',
-    '```mermaid',
-    'graph LR;',
+    "```",
+    "",
+    "## all dependencies",
+    "```mermaid",
+    "graph LR;",
     ...annotationsAll.sort(),
-    '```',
-    '',
-    '## npm dependency hierarchy',
-    '',
-    'These are the groups of dependencies in npm that depend on each other.',
-    'Each group depends on packages lower down the chain, nothing depends on',
-    'packages higher up the chain.',
-    '',
-    ` - ${hierarchyOurs.reverse().join('\n - ')}`,
-  ]
-
-  return fs.writeFile(join(CWD, 'DEPENDENCIES.md'), out.join('\n'))
+    "```",
+    "",
+    "## npm dependency hierarchy",
+    "",
+    "These are the groups of dependencies in npm that depend on each other.",
+    "Each group depends on packages lower down the chain, nothing depends on",
+    "packages higher up the chain.",
+    "",
+    ` - ${hierarchyOurs.reverse().join("\n - ")}`,
+  ];
+
+  await writeFile(join(__dirname, "DEPENDENCIES.md"), out.join("\n"));
 }
 
-const walk = function (tree, onlyOurs) {
-  const annotations = [] // mermaid dependency annotations
-  const dependedBy = {}
+function walk(tree, onlyOurs) {
+  const annotations = [];
+  const dependedBy = {};
 
-  iterate(tree, dependedBy, annotations, onlyOurs)
+  iterate(tree, dependedBy, annotations, onlyOurs);
 
-  const allDeps = new Set(Object.keys(dependedBy))
-  const foundDeps = new Set()
-  const hierarchy = []
+  const allDeps = new Set(Object.keys(dependedBy));
+  const foundDeps = new Set();
+  const hierarchy = [];
 
   if (onlyOurs) {
     while (allDeps.size) {
-      log.silly('SIZE', allDeps.size)
```

Comment:
Logging is important here as we are dealing with some deeply nested tree traversal.